src/sentry/api/endpoints/sentry_app_publish_request.py

```diff
@@ -19,6 +27,21 @@ def post(self, request, sentry_app):
         if sentry_app.is_publish_request_inprogress:
             return Response({"detail": "Publish request in progress."}, status=400)
 
+        if features.has("organizations:sentry-app-logo-upload", sentry_app.owner):
+            if not SentryAppAvatar.objects.filter(sentry_app=sentry_app, color=True).exists():
```

Comment:
This is probably overkill but you can do both of these `exists()` checks in a single query:
```py
avatar_counts_by_color = {
    row["color"]: row["color_count"]
    for row in (
        SentryAppAvatar.objects.filter(sentry_app=sentry_app)
        .values("color")
        .annotate(color_count=Count("color"))
    )
}

if not avatar_counts_by_color.get(True):
    ...
```